lucene/sandbox/src/java/org/apache/lucene/geo/Rectangle2D.java

```diff
@@ -148,6 +148,90 @@ public boolean intersectsTriangle(int aX, int aY, int bX, int bY, int cX, int cY
+    if (this.crossesDateline() == true) {
+      //Triangles cannot cross the date line so it is always false
+      return EdgeTree.WithinRelation.CROSSES;
+    }
+    // compute bounding box of triangle
+    int tMinX = StrictMath.min(StrictMath.min(ax, bx), cx);
+    int tMaxX = StrictMath.max(StrictMath.max(ax, bx), cx);
+    int tMinY = StrictMath.min(StrictMath.min(ay, by), cy);
+    int tMaxY = StrictMath.max(StrictMath.max(ay, by), cy);
+    if (boxesAreDisjoint(tMinX, tMaxX, tMinY, tMaxY,  minX, maxX, minY, maxY)) {
+      return EdgeTree.WithinRelation.DISJOINT;
+    }
+
+    return bboxWithinTriangle(ax, ay, ab, bx, by, bc, cx, cy, ca, minX, maxX, minY, maxY);
+  }
+  public EdgeTree.WithinRelation bboxWithinTriangle(int ax, int ay, boolean ab, int bx, int by, boolean bc, int cx, int cy, boolean ca, int minLon, int maxLon, int minLat, int maxLat) {
+    //points belong to the shape so if points are inside the rectangle then it cannot be within. It is valid if the point
+    //is on the edge.
+    if (bboxContainsPoint(ax, ay, minLon, maxLon, minLat, maxLat) ||
+        bboxContainsPoint(bx, by, minLon, maxLon, minLat, maxLat) ||
+        bboxContainsPoint(cx, cy, minLon, maxLon, minLat, maxLat)) {
+      return EdgeTree.WithinRelation.CROSSES;
+    }
+    //if any of the edges crosses and edge belonging to the polygon then it cannot be within.
+    //Note that crosses is different to crosses as it needs to have points at both sides.
+    boolean inside = false;
+    if  (edgeIntersectsBox(ax, ay, bx, by, minLon, maxLon, minLat, maxLat) == true) {
+      if (ab == true) {
+        return EdgeTree.WithinRelation.CROSSES;
+      } else {
+        inside = true;
+      }
+    }
+    if (edgeIntersectsBox(bx, by, cx, cy, minLon, maxLon, minLat, maxLat) == true) {
+      if (bc == true) {
+        return EdgeTree.WithinRelation.CROSSES;
+      } else {
+        inside = true;
+      }
+    }
+    if (edgeIntersectsBox(cx, cy, ax, ay, minLon, maxLon, minLat, maxLat) == true) {
+      if (ca == true) {
+        return EdgeTree.WithinRelation.CROSSES;
+      } else {
+        inside = true;
+      }
+    }
+    //if any of the edges crosses and edge that does not belong to the polygon
+    // then it is a candidate for within
+    if (inside == true) {
+      return EdgeTree.WithinRelation.CANDIDATE;
+    }
+
+    // Check if the rectangle is fully within the triangle, if not then the relationship is undefined.
+    // compute bounding box of triangle
+    int tMinX = StrictMath.min(StrictMath.min(ax, bx), cx);
+    int tMaxX = StrictMath.max(StrictMath.max(ax, bx), cx);
+    int tMinY = StrictMath.min(StrictMath.min(ay, by), cy);
+    int tMaxY = StrictMath.max(StrictMath.max(ay, by), cy);
+    if ((tMinX <= minLon && tMaxX >= maxLon && tMinY <= minLat && tMaxY >= maxLat) == false) {
+      return EdgeTree.WithinRelation.DISJOINT;
+    }
```

Comment:
we don't seem to need this as this is only called from withinTriangle which checks this for us?